app/client/src/widgets/TabsWidget/widget/index.tsx

```diff
@@ -328,11 +328,18 @@ class TabsWidget extends BaseWidget<
           this.props.updateWidgetMetaProperty("selectedTabWidgetId", undefined);
         }
       } else if (!this.props.selectedTabWidgetId) {
-        if (visibleTabs.length > 0) {
-          this.props.updateWidgetMetaProperty(
-            "selectedTabWidgetId",
-            visibleTabs[0].widgetId,
-          );
+        if (Object.keys(this.props.tabsObj || {}).length) {
+          // If we have a defaultTab
+          if (this.props.defaultTab) {
+            this.setDefaultSelectedTabWidgetId();
+          } else {
+            // If no tab is selected
+            // Select the first tab in the tabs list.
+            this.props.updateWidgetMetaProperty(
+              "selectedTabWidgetId",
+              visibleTabs.length ? visibleTabs[0].widgetId : undefined,
```

Comment:
We could replace this with
```suggestion
              visibleTabs?.[0]?.widgetId,
```